### PR TITLE
Fix multiple typos in C files

### PR DESCRIPTION
**Pull request by atoomic**

### CODE REVIEW

File: charclass_invlists.h [hunk 1]
```diff
@@ -434684,7 +434684,7 @@ static const char * const UNI_lb_values[] = {
 	"infixnumeric",
 	"Inseparable",
 	"inseparable",
-	"Inseperable",
+	"Inseparable",
```
Comment (tonycoz) [inline charclass_invlists.h]: a) this file is generated
b) the mis-spelling here is specified by [Unicode](http://www.unicode.org/Public/UCD/latest/ucd/PropertyValueAliases.txt).  Note the correct spelling is also available.

File: regexec.c [hunk 2]
```diff
@@ -3583,7 +3583,7 @@ Perl_regexec_flags(pTHX_ REGEXP * const rx, char *stringarg, char *strend,
     regnode *c;
     char *startpos;
     SSize_t minlen;		/* must match at least this many chars */
-    SSize_t dontbother = 0;	/* how many characters not to try at end */
+    SSize_t don'tbother = 0;	/* how many characters not to try at end */
```
Comment (tonycoz) [inline regexec.c]: This is wrong, C doesn't allow `'` in variable names.

Further down too.

File: uconfig.h [hunk 3]
```diff
@@ -3594,7 +3594,7 @@
 
 /* HAS_WCSXFRM:
  *	This symbol, if defined, indicates that the wcsxfrm routine is
- *	available to tranform a wide character string for wcscmp().
+ *	available to transform a wide character string for wcscmp().
```
Comment (tonycoz) [inline uconfig.h:3597]: This change should be made in `config_h.SH`.  This change will be overridden on a make regen.

File: win32/perlhost.h [hunk 4]
```diff
@@ -108,7 +108,7 @@ class CPerlHost
     };
 
 /* IPerlMemParse */
-    /* Assume something else is using locks to mangaging serialize
+    /* Assume something else is using locks to manage serialize
        on a batch basis
```
Comment (tonycoz) [inline win32/perlhost.h]: serialization?

File: charclass_invlists.h (hunk 1 above)
Comment (atoomic) [reply charclass_invlists.h]: change reverted, thanks

File: regexec.c (hunk 2 above)
Comment (atoomic) [reply regexec.c]: indeed... I should have seen it. Fixed

File: uconfig.h (hunk 3 above)
Comment (atoomic) [reply uconfig.h:3597]: updated with other generated files

File: win32/perlhost.h (hunk 4 above)
Comment (atoomic) [reply win32/perlhost.h]: applied the suggestion

**Comment by tonycoz** [comment]
Please `make test_porting` before pushing again.

File: uconfig.h (hunk 3 above)
Comment (Tux) [reply uconfig.h:3597]: I updated this in `metaconfig/U/perl/d_wcsxfrm.U` as 87e76cbf7e9ee95ecd1bfe81e6ddd31351ada25f
will be included after the next regen

**Comment by jkeenan** [comment]
Superseded by https://github.com/Perl/perl5/pull/20646, which was merged into blead in 953398c3fd421376c66841d7658f8b97328e2726 (Dec 29 2022); closing.

**Pull request closed without merging**